src/views/Account/Rewards/RewardsCard.tsx

```diff
@@ -0,0 +1,834 @@
+          />
+        )
+      })}
+    </PromotionList>
+  )
+}
+
+// PREDICTION / ESTIMATE:
+// (user twab balance for epoch / twab total supply for epoch) * tokensPerEpoch
+//
+// my twab: 200 for epoch 1
+// twab total supply (currently for 1 chain): 1000 for epoch 1
+//
+// 200/1000 (or 20%) is my vApr
+// 30 tokens given away for epoch 1
+// = I get 6 tokens for epoch 1
+//
+// remaining epochs: 8
+// 6 * 8 = 48
+// I'll get 48 tokens over the entire time (if nothing changes)
+const PromotionRow = (props) => {
+  const { promotion, chainId, refetchUsersRewardsHistory } = props
+  const { id, maxCompletedEpochId, token: tokenAddress } = promotion
+
+  const [isOpen, setIsOpen] = useState(false)
+
+  const { data: token, isFetched: tokenIsFetched } = useToken(chainId, tokenAddress)
+
+  const usersAddress = useUsersAddress()
+
+  const { data: usersPromotionData, refetch: refetchUsersRewardsAmount } =
+    useUsersPromotionRewardsAmount(chainId, Number(id), maxCompletedEpochId, usersAddress)
+
+  const { id: promotionId } = promotion
+
+  const {
+    data: claimable,
+    isFetched: claimableIsFetched,
+    refetch: refetchClaimable
+  } = useUsersPromotionAmountClaimable(chainId, promotionId, usersPromotionData, token)
+  const { amount: claimableAmount, usd: claimableUsd } = claimable || {}
+
+  const {
+    data: estimate,
+    isFetched: estimateIsFetched,
+    refetch: refetchEstimate
+  } = useUsersPromotionAmountEstimate(chainId, promotion, token)
+  const { amount: estimateAmount, usd: estimateUsd } = estimate || {}
+
+  const total = Number(estimateAmount?.amount) + Number(claimableAmount?.amount)
+  const totalUsd = estimateUsd + claimableUsd
+
+  const refetch = () => {
+    refetchUsersRewardsHistory()
+    refetchUsersRewardsAmount()
+    refetchEstimate()
+    refetchClaimable()
+  }
+
+  return (
+    <>
+      {!tokenIsFetched ? (
+        <RewardsCardLoadingList listItems={1} />
+      ) : (
+        <>
+          <PromotionListItem
+            onClick={() => {
+              setIsOpen(true)
+            }}
+            left={
+              <div className='flex items-center font-bold'>
+                <img className='w-5 mr-2' src='beach-with-umbrella.png' /> {token.symbol}{' '}
+              </div>
+            }
+            right={
+              <div className='flex items-center'>
+                <TokenIcon
+                  chainId={chainId}
+                  address={token?.address}
+                  sizeClassName='w-5 h-5'
+                  className='mr-2'
+                />
+                <RewardsBalance
+                  estimateUsd={estimateUsd}
+                  claimableUsd={claimableUsd}
+                  isFetched={claimableIsFetched && estimateIsFetched}
+                />{' '}
+                <FeatherIcon icon='chevron-right' className='my-auto w-6 h-6 opacity-50' />
+              </div>
+            }
+          />
+
+          <ClaimModal
```

Comment:
This follows the pattern I did for balances which is actually kinda bad. It results in a Modal for every row which gets pretty laggy - I've been switching to a different pattern more like the delegation tool with a single modal that all of the rows can open & close